Code::Blocks project: use the msys2's wxWidgets library to build the …

asmwarrior opened the pull request:
…DXF_Viewer

The compiler name is changed to "gcc".
The wxWidgets library can be obtained by the pacman command, see here: Base Package: mingw-w64-wxwidgets3.2 - MSYS2 Packages — https://packages.msys2.org/base/mingw-w64-wxwidgets3.2 Use the wx-config-msys2.exe to generate the correct compiler options. This tool can be obtained from https://github.com/eranif/wx-config-msys2 Adjust the file path in the cbp.
Add a resource rc file for the project.
I don't think the "StdAfx.h"(the PCH file name for MSVC) file is needed, so I remove it from the cbp file.

asmwarrior committed:
Code::Blocks project: use the msys2's wxWidgets library to build the DXF_Viewer

The compiler name is changed to "gcc".
The wxWidgets library can be obtained by the pacman command, see here: Base Package: mingw-w64-wxwidgets3.2 - MSYS2 Packages — https://packages.msys2.org/base/mingw-w64-wxwidgets3.2
Use the wx-config-msys2.exe to generate the correct compiler options. This tool can be obtained from https://github.com/eranif/wx-config-msys2
Adjust the file path in the cbp.
I don't think the "StdAfx.h"(the PCH file name for MSVC) file is needed, so I remove it from the cbp file.

asmwarrior committed:
use code::blocks' gcv(global compiler variable)

In this case, we can either use msys2's wx library or self build wx library,
see discussion here:
Re: New Variable to support wx-config in MSys2 MinGW
https://forums.codeblocks.org/index.php/topic,24974.msg176013.html#msg176013

JamesBremner commented on src/resource.rc:
```diff
@@ -0,0 +1,3 @@
+aaaa ICON "wx/msw/std.ico"
```
This seems very specific to wxWidgets and codeblocks.

Can you isolate this from the other builds?  If it is accidentally included in a build on machines without wxWidgets installed there will be no `wx/msw` folders and the result will be mysterious build errors.

If it absolutely has to be in the src folder, can you at least add some comment describing the purpose of this file and what to do if it causes problems?

asmwarrior committed:
remove the wx related code in Dxf.h

see discussion here:
Limit GUI framework dependencies to one file · Issue #35 · JamesBremner/DXF_Viewer — https://github.com/JamesBremner/DXF_Viewer/issues/35

asmwarrior commented:
I just add two extra commits to implement the #35 

I will do other stuff tomorrow, good night.

JamesBremner commented:
> I just add two extra commits to implement the https://github.com/JamesBremner/DXF_Viewer/issues/35

That is a pity.  The changes for #35 look fine.  If they were in their own pull request I would accept them.

There is a problem with the commit of resource.rc to the src folder which prevent me from accepting the entire pull request.

JamesBremner commented on src/wxWidgets.cpp:
```diff
@@ -242,7 +246,7 @@ void dxfv_wxWidgetsFrame::OnSelectSolidParser(wxCommandEvent& event)
 {
     wxPaintDC dc(this);
-    dxf->DC( &dc );
+    theDC = &dc; // set the global wxDC pointer
```
Is this a memory leak?

The pointer to the old DC is being overwritten by the pointer to the new DC.  So this will work until memory is exhausted.  This could be quite quickly if the pan and zoom features are used frequently.  AFAIK the old DC should be deleted before the pointer is overwritten.

JamesBremner replied:
Oops, my mistake.  dc is an automatic variable, so it is deleted when OnPaint() exits.  Everything is fine.

If you create a pull request with just this commit, I will accept it.

asmwarrior commented:
OK.

I have removed the rc file, because I think this icon is not that necessary

I have change all the commits(interactive git rebase), and rebased on your current master, and make a push with force option.
